server/src/pdns.rs

```diff
@@ -290,14 +306,12 @@ fn process_request(req: PdnsRequest, config: &Config) -> Result<PdnsResponse, St
                         .result
                         .push(PdnsResponseParams::Lookup(ns_record));
                 }
-
-                return Ok(pdns_response);
             }
             Err(_) => {
-                // No such domain, return a `false` result to PowerDNS.
-                return Err("No record for this name.".to_owned());
+                info!("No record for this name {}", qname);
```

Comment:
I am not convinced that this correct to not send an error here anymore. Can you explain the logic change here?

Reply:
We need an MX record to always be responded (since we need to receive emails in the tld) even if the qname does not exist in the database. 

The situation I faced when testing while having this error being returned to pdns, was that when an MX query was made, it never responded the DNS query properly, because since the qname wasn't found in the db, we had `Err("No record for this name.".to_owned());` always being sent in the end, instead of `Ok(pdns_response);` consequently making the SMTP unable to reach the MX server and ultimately returning the email.

If I had at least the correct query type being attributed to`qtype`  (it always only have `SOA` and `ANY`) I could overcome this and return immediately if the request is an `MX` but I'm not sure if something need to be changed on powerdns or if that's the normal behavior. 

This approach was tested on staging and didn't affect neither powerdns, pagekite and the API.

Reply:
This approach was tested on staging and didn't affect neither powerdns, pagekite and the API.

Reply:
Another situation we faced that illustrates this use case was we needed to add a TXT challenge record to the TLD to allow amazon to handle the domain's SMTP. With the  `Err("No record for this name.".to_owned());` being returned when the `qname` isn't found we never had the challenge validated because nothing except an error was returned all the time.